### PR TITLE
Use Quill editor instead of MediumEditor

### LINKED ISSUE #877: Can't upload image to editor in Firefox

### LINKED ISSUE #922: Using quill (editor)
https://github.com/quilljs/quill

### PR DESCRIPTION
**Pull request by ArsalanSavand**
Fixes #922 
Fixes #877 

- Normal mode
image

- Dark mode
image

### CODE REVIEW

**Comment by ArsalanSavand** [comment]
@Alir3z4 and @AmirSavand, checkout to branch and see it.

File: src/app/entry/entry_edit/entry_edit_controller.js [hunk 1]
```diff
@@ -96,6 +93,15 @@ function EntryEditController($scope, $rootScope, $state, $stateParams, $timeout,
         );
       }, 1000);
     }
+
+    // $scope.options = {
```
Comment (Alir3z4) [inline src/app/entry/entry_edit/entry_edit_controller.js]: You may just delete this commented code

**Comment by Alir3z4** [comment]
I'm going to try it on my local computer.

**Comment by AmirSavand** [comment]
@Alir3z4 check the functionality, this pull is not ready to be merged. Design is not done and image uploading is not implemented.

**Comment by Alir3z4** [comment]
Ok, I see.
I'm looking into it to see how it is

**Comment by Alir3z4** [comment]
Before merging this also we need to do the lazy loading to load the assets
only in the editor page

**Comment by Alir3z4** [comment]
OK, lazy loading part is done.
Please go on and have fun, keep stabbing until it ends

**Comment by AmirSavand** [comment]
@Alir3z4, @ArsalanSavand can't find a way to handle image selection.

**Comment by Alir3z4** [comment]
These are the stuff you're looking for:

* https://quilljs.com/docs/modules/toolbar/#handlers
* https://github.com/quilljs/quill/issues/1089: Read the comment thread now https://github.com/quilljs/quill/issues/1089 it contains examples from other people whe had similar problem to fix and some of them provided their solution

**Comment by ArsalanSavand** [comment]
@Alir3z4 we are not using quilljs, we are using ngQuill.
ngQuill is the angularjs version of quilljs

**Comment by ArsalanSavand** [comment]
It doesn't have too much handlers

**Comment by Alir3z4** [comment]
We're using quilljs, ngQuill is just a wrapper around it.
@AmirSavand can you look into this ?

**Comment by Alir3z4** [comment]
@ArsalanSavand have a look at this and let me know https://github.com/KillerCodeMonkey/ng-quill/issues/116

File: src/app/entry/entry_edit/entry_edit_controller.js [hunk 2]
```diff
@@ -147,14 +147,32 @@ function EntryEditController($scope, $rootScope, $state, $stateParams, $timeout,
...
     };
   }
 
+  $scope.onEditorInit = function(editor) {
+    $scope.editor = editor;
+    $scope.cursorIndex = 0;
+
+    let toolbar = editor.getModule('toolbar');
+    toolbar.addHandler('image', () => {
+      let range = editor.getSelection();
```
Comment (Alir3z4) [inline src/app/entry/entry_edit/entry_edit_controller.js]: Couldn't this be `const` instead `let` ?

File: src/style/css/main.css [hunk 3]
```diff
@@ -2071,3 +1925,219 @@ img.cache {
...
+    color: #d17100 !important;
+
+}
+
+.ql-snow .ql-picker-options .ql-picker-item:hover {
+    background-color: #eee;
+    color: #d17100 !important;
+}
+
+.ql-snow.ql-toolbar button:hover .ql-stroke,
```
[the remark this answers is not in the document]
Comment (Alir3z4) [reply src/style/css/main.css]: When talking with @AmirSavand about SCSS it's like talking he never ever heard about it.
See this thread https://github.com/SavandBros/gonevis-dash/issues/916 it's like talking with a wall.

File: src/style/css/main.css [hunk 4]
```diff
@@ -2071,3 +1925,219 @@ img.cache {
     visibility: visible;
     opacity: 1;
 }
+
+
+/* Editor */
```
[the remark this answers is not in the document]
Comment (ArsalanSavand) [reply src/style/css/main.css]: Done

[the remark this answers is not in the document]
Comment (Alir3z4) [reply src/style/css/main.css]: Excellent!

**Comment by Alir3z4** [comment]
@ArsalanSavand anything left ?
can we merge ?

**Comment by ArsalanSavand** [comment]
@Alir3z4  Good to merge

**Pull request merged**